### PR TITLE
Add GPO list daemon timeout option

### PR DESCRIPTION
**Pull request by dangerousplay**
Hello, I hope you are good.

Added a `--gpo-list-timeout` flag and a `gpo_list_timeout` configuration parameter on the adsys daemon.

By default, the GPO list timeout is fixed to 10 seconds. If the GPO list python script takes more than 10 seconds it's killed by the timeout. However, there might be scenarios where 10 seconds might not be enoguth. In our use case, the GPO list procedure finishes around 15 seconds in average. 

Have a wonderful day ^^

### CODE REVIEW

**Review by didrocks — CHANGES_REQUESTED** [review]
Excellent! Thanks for contributing to adsys and help making it better (and work in your case :)). I think the approach makes sense. You are the first one enduring this timeout. We may increase the default later if more cases occurred, but let’s keep it that way with your configuration option to accomodate other scenarios for now.

I have one small nitpick in the example configuration file. We can modify it after merging if you don’t have time to modify it. It’s up to you, just tell us :)
Then, we can merge.

Thanks again for this very valuable contribution!

File: conf.example/adsys.yaml [hunk 1]
```diff
@@ -36,3 +36,6 @@ detect_cached_ticket: false
 
 # Client only configuration
 client_timeout: 60
+
+# GPO List timeout
+gpo_list_timeout: 30
```
[the remark this answers is not in the document]
Comment (dangerousplay) [reply conf.example/adsys.yaml:41]: Changed to 10 seconds to keep consistent with the current default value ^^

**Comment by dangerousplay** [comment]
Thank you, pull request updated with the suggestion ^^

**Review by didrocks — APPROVED** [review]
Thanks a lot for you work! I just pushed one commit to fix the capitalization of "gpo" in the help, as the generated helper from the reference command line was making the documentation check (so that we don’t have a mix of "GPO" and "gpo" in it). Howver, because of the configuration using "gpo", I had to make it enter the allowed list too. I didn’t want you to have to bother with this :)

Everything looks excellent! Approving and merging, thanks again :)

**Pull request merged**